homeassistant/components/hvac/__init__.py

```diff
@@ -0,0 +1,492 @@
+        """Set humidity on the target hvacs."""
+        target_hvacs = component.extract_from_service(service)
+
+        humidity = util.convert(
+            service.data.get(ATTR_HUMIDITY), float)
+
+        if humidity is None:
+            _LOGGER.error(
+                "Received call to %s without attribute %s",
+                SERVICE_SET_HUMIDITY, ATTR_HUMIDITY)
+            return
+
+        for hvac in target_hvacs:
+            hvac.set_humidity(humidity)
+
+            hvac.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_HUMIDITY, humidity_set_service,
+        descriptions.get(SERVICE_SET_HUMIDITY))
+
+    def fan_mode_set_service(service):
+        """Set fan mode on target hvacs."""
+        target_hvacs = component.extract_from_service(service)
+
+        fan_mode = service.data.get(ATTR_FAN_MODE)
+
+        if fan_mode is None:
+            _LOGGER.error(
+                "Received call to %s without attribute %s",
+                SERVICE_SET_FAN_MODE, ATTR_FAN_MODE)
+            return
+
+        for hvac in target_hvacs:
+            hvac.set_fan_mode(fan_mode)
+
+            hvac.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_FAN_MODE, fan_mode_set_service,
+        descriptions.get(SERVICE_SET_FAN_MODE))
+
+    def operation_set_service(service):
+        """Set operating mode on the target hvacs."""
+        target_hvacs = component.extract_from_service(service)
+
+        operation_mode = service.data.get(ATTR_OPERATION)
+
+        if operation_mode is None:
+            _LOGGER.error(
+                "Received call to %s without attribute %s",
+                SERVICE_SET_OPERATION_MODE, ATTR_OPERATION)
+            return
+
+        for hvac in target_hvacs:
+            hvac.set_operation(operation_mode)
+
+            hvac.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_OPERATION_MODE, operation_set_service,
+        descriptions.get(SERVICE_SET_OPERATION_MODE))
+
+    def swing_set_service(service):
+        """Set swing mode on the target hvacs."""
+        target_hvacs = component.extract_from_service(service)
+
+        swing_mode = service.data.get(ATTR_SWING_MODE)
+
+        if swing_mode is None:
+            _LOGGER.error(
+                "Received call to %s without attribute %s",
+                SERVICE_SET_SWING, ATTR_SWING_MODE)
+            return
+
+        for hvac in target_hvacs:
+            hvac.set_swing(swing_mode)
+
+            hvac.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_SWING, swing_set_service,
+        descriptions.get(SERVICE_SET_SWING))
+    return True
+
+
+class HvacDevice(Entity):
+    """Representation of a hvac."""
+
+    # pylint: disable=too-many-public-methods,no-self-use
+    @property
+    def state(self):
+        """Return the current state."""
+        return self.target_temperature or STATE_UNKNOWN
```

Comment:
I think the state should not be the temperature but the current mode: is it heating? cooling ?